### PR TITLE
Feature: add coverage report

### PR DESCRIPTION
**Pull request by Olavhaasie**
Partly closes PM-79

## Description
Added codecov support for this repository. The [reports](https://codecov.io/gh/project-march/monitor/branch/feature%2FPM-79-add-coverage-report) show covered lines and branch coverage. Travis will upload the coverage report when the build succeeded.

## Changes
* Add coverage report uploading to Travis build
* Add tests to increase coverage

### CODE REVIEW

File: march_rqt_note_taker/test/entry_model_test.py [hunk 1]
```diff
@@ -14,10 +15,20 @@ def test_empty_model(self):
...
     def test_insert_single_row(self):
-        self.model.insert_row(Entry('test'))
+        entry = Entry('test', QDateTime.fromSecsSinceEpoch(5))
+        self.model.insert_row(entry)
         self.assertEqual(self.model.rowCount(), 1)
 
+        time_index = self.model.createIndex(0, 0)
+        content_index = self.model.createIndex(0, 1)
+        self.assertEqual(self.model.data(time_index), entry.time_string())
+        self.assertEqual(self.model.data(content_index), entry.content)
```
Comment (Roelemans) [inline march_rqt_note_taker/test/entry_model_test.py:34]: Should you not limit tests to one assert per test?

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[report]
+include =
+    *march_rqt_gait_selection*
+    *march_rqt_input_device*
+    *march_rqt_note_taker*
+omit =
+    *test*
```
Comment (Joris-JW) [inline .coveragerc]: multiple example show that *test* is not included in the coverage report. Isn't it useful to check whether the coverage of your test is 100%?

**Review by Joris-JW — COMMENTED** [review]
Looks very nice! just a small question added

File: .coveragerc (hunk 2 above)
Comment (Olavhaasie) [reply .coveragerc]: I will check if I can split the test coverage from the actual coverage as to not get skewed results when test coverage is included

File: march_rqt_note_taker/test/entry_model_test.py (hunk 1 above)
Comment (Olavhaasie) [reply march_rqt_note_taker/test/entry_model_test.py:34]: I will move the bottom part that I added to a separate test.

File: .coveragerc (hunk 2 above)
Comment (Olavhaasie) [reply .coveragerc]: The coverage of the tests are now also included but are not used in the PR comment, commit status and badge :tada:

File: march_rqt_note_taker/src/march_rqt_note_taker/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from .entry import Entry
+from .entry_model import EntryModel
+from .filter_map import FilterMap
+from .notes_plugin import NotesPlugin
+from .notes_widget import NotesWidget
```
Comment (Roelemans) [inline march_rqt_note_taker/src/march_rqt_note_taker/__init__.py]: Can this be moved to the __init__ in the test folder?

**Review by Roelemans — CHANGES_REQUESTED** [review]

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+coverage:
+  status:
+    project:
+      default: off
+      production:
+        target: auto
```
Comment (Joris-JW) [inline codecov.yml:6]: why use auto? Should we not set a coverage number (maybe with a certain tolerance) from now on?

Comment (Olavhaasie) [reply codecov.yml:6]: Auto uses the target branch coverage as target. So the coverage is not allowed to be lower. Setting this value to some value like 80% will cause the check to fail everytime until we have added enough tests.

File: march_rqt_note_taker/src/march_rqt_note_taker/__init__.py (hunk 3 above)
Comment (Olavhaasie) [reply march_rqt_note_taker/src/march_rqt_note_taker/__init__.py]: I think so, let me try

Comment (Olavhaasie) [reply march_rqt_note_taker/src/march_rqt_note_taker/__init__.py]: It worked :+1:

**Review by Roelemans — APPROVED** [review]
Nice :)

File: codecov.yml (hunk 4 above)
Comment (Joris-JW) [reply codecov.yml:6]: So if we add extra code additional test **have**  to be written otherwise the coverage report would be below the target branch?

**Review by Joris-JW — COMMENTED** [review]
again, PR looks fine just an additional question.

File: codecov.yml (hunk 4 above)
Comment (Olavhaasie) [reply codecov.yml:6]: Yes, so If you add new code you add new tests for that new code

**Review by Joris-JW — APPROVED** [review]

**Pull request merged**